test/ref_ops_test.cpp

```diff
@@ -3593,4 +3593,369 @@ TEST_CASE(unsqueeze_test)
+                                1.0,
+                                2.0,
+                                3.0,
+                                2.0,
+                                4.0,
+                                6.0,
+                                0.0,
+                                0.0,
+                                0.0,
+                                1.0,
+                                2.0,
+                                3.0,
+                                2.0,
+                                4.0,
+                                6.0};
+        EXPECT(results_vector == gold);
+    }
+}
+
+TEST_CASE(prefix_scan_sum_reverse)
+{
+    {
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        migraphx::shape s{migraphx::shape::float_type, {8}};
+        auto input = migraphx::literal{s, {1, 2, 3, 4, 1, 2, 3, 4}};
+        auto l0    = mm->add_literal(input);
+        mm->add_instruction(
+            migraphx::make_op("prefix_scan_sum",
+                              {{"axis", 0}, {"exclusive", false}, {"reverse", true}}),
+            l0);
+        p.compile(migraphx::ref::target{});
+        auto result = p.eval({}).back();
+        std::vector<float> results_vector;
+        result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+        std::vector<float> gold{20.0, 19.0, 17.0, 14.0, 10.0, 9.0, 7.0, 4.0};
+        EXPECT(results_vector == gold);
+    }
+
+    {
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        migraphx::shape s{migraphx::shape::float_type, {2, 2, 2}};
+        auto input = migraphx::literal{s, {1, 2, 3, 4, 1, 2, 3, 4}};
+        auto l0    = mm->add_literal(input);
+        mm->add_instruction(
+            migraphx::make_op("prefix_scan_sum",
+                              {{"axis", 0}, {"exclusive", false}, {"reverse", true}}),
+            l0);
+        p.compile(migraphx::ref::target{});
+        auto result = p.eval({}).back();
+        std::vector<float> results_vector;
+        result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+        std::vector<float> gold{2.0, 4.0, 6.0, 8.0, 1.0, 2.0, 3.0, 4.0};
+        EXPECT(results_vector == gold);
+    }
+}
+
+TEST_CASE(prefix_scan_sum_exclusive_reverse)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    migraphx::shape s{migraphx::shape::float_type, {6}};
+    auto input = migraphx::literal{s, {1, 2, 3, 4, 5, 6}};
+    auto l0    = mm->add_literal(input);
+    mm->add_instruction(
+        migraphx::make_op("prefix_scan_sum", {{"axis", 0}, {"exclusive", true}, {"reverse", true}}),
+        l0);
+    p.compile(migraphx::ref::target{});
+    auto result = p.eval({}).back();
+    std::vector<float> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+    std::vector<float> gold{20.0, 18.0, 15.0, 11.0, 6.0, 0.0};
+    EXPECT(results_vector == gold);
+}
+
```

Comment:
Now the unit test are sorted in the alphabetical order, can you move the tests so they are in order in the file?

Reply:
Fixed